modules/nf-core/bcftools/pluginimputeinfo/main.nf

```diff
@@ -0,0 +1,76 @@
+process BCFTOOLS_PLUGINIMPUTEINFO {
+    tag "$meta.id"
+    label 'process_single'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/bcftools:1.20--h8b25389_1':
+        'biocontainers/bcftools:1.20--h8b25389_1' }"
+
+    input:
+    tuple val(meta), path(vcf), path(index)
+    path(regions)
+    path(targets)
+
+    output:
+    tuple val(meta), path("*.{vcf,vcf.gz,bcf,bcf.gz}"), emit: vcf
+    tuple val(meta), path("*.tbi")                    , emit: tbi, optional: true
+    tuple val(meta), path("*.csi")                    , emit: csi, optional: true
+    path "versions.yml"                               , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def args2 = task.ext.args2 ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def regions_file = regions ? "--regions-file ${regions}" : ""
+    def targets_file = targets ? "--targets-file ${targets}" : ""
+    def extension = args.contains("--output-type b") || args.contains("-Ob") ? "bcf.gz" :
+                    args.contains("--output-type u") || args.contains("-Ou") ? "bcf" :
+                    args.contains("--output-type z") || args.contains("-Oz") ? "vcf.gz" :
+                    args.contains("--output-type v") || args.contains("-Ov") ? "vcf" :
+                    "vcf"
+    """
+    bcftools plugin impute-info \\
+        --output ${prefix}.${extension} \\
+        ${regions_file} \\
+        ${targets_file} \\
+        $args \\
+        --threads $task.cpus \\
+        ${vcf} \\
+        -- \\
+        $args2
```

Comment:
What is the purpose of this second args? There is only one tool

Reply:
BCFtools plugins are run as `bcftools [plugin NAME|+NAME] [OPTIONS] FILE  --  [PLUGIN OPTIONS]`. Most plugins accept two types of parameters: general options shared by all plugins followed by a separator, and a list of plugin-specific options. I've set `args1` to be the general options and `args2` to be the plugin-specific options. For this particular plugin there are not any plugin-specific options so `args2` isn't currently necessary but I'm not sure whether to remove it since it is not being used or to keep it in case a future update does include plugin-specific options. Would you mind sharing your thoughts on this? @jfy133

Reply:
Ahh I see! Your logic makes sense indeed, I think you should somehow document it in the meta.yaml as this isn't maybe strictly the standard way of doing this